### PR TITLE
prevent dismissing the loading spinner on a redirect

### PR DESCRIPTION
**Pull request by kiftio**
### What are you trying to accomplish?

`onPageFinished` is being invoked twice on a normal guest flow, the first appears to be the cart redirect. This is causing us to dismiss the loading spinner too early, and send an instrumentation message twice:

https://github.com/Shopify/checkout-sheet-kit-android/assets/15106863/0f46c29e-011c-41fe-baab-158648cf71f9

URLs:
- `https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/cart/c/c1-cee737be76e498a6eefa107843517818?key=0f92b50b0375b1c8224d0950e0bb5210`
- `https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/checkouts/cn/c1-cee737be76e498a6eefa107843517818?locale=en-CA&skip_shop_pay=prompt_allowed`

This PR adds a condition, that we should not set loadComplete to true for `onPageFinished` calls if the content height of the view is 0.

https://github.com/Shopify/checkout-sheet-kit-android/assets/15106863/9265e005-6467-45b2-b7b6-7b2e4adff2c8

### Before you deploy

- [x] I have added tests to support my implementation
- [x] I have read and agree with the [contributing documentation](https://github.com/Shopify/checkout-sheet-kit-android/blob/main/.github/CONTRIBUTING.md)
- [x] I have read and agree with the [code of conduct documentation](https://github.com/Shopify/checkout-sheet-kit-android/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I have updated any documentation related to these changes.
- [ ] I have updated the [README](https://github.com/Shopify/checkout-sheet-kit-android/blob/main/README.md) (if applicable).

### CODE REVIEW

File: lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt [hunk 1]
```diff
@@ -138,6 +138,8 @@ internal class CheckoutWebView(context: Context, attributeSet: AttributeSet? = n
 
         override fun onPageFinished(view: WebView, url: String) {
             super.onPageFinished(view, url)
+            if (view.contentHeight == 0) return
```
Comment (markmur) [inline lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt:141]: what exactly is this checking? 🤔

Comment (kiftio) [reply lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt:141]: If you look at the URLs in the description:

1. https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/cart/c/c1-cee737be76e498a6eefa107843517818?key=0f92b50b0375b1c8224d0950e0bb5210
2. https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/checkouts/cn/c1-cee737be76e498a6eefa107843517818?locale=en-CA&skip_shop_pay=prompt_allowed

The first looks like the cart -> checkout redirect to me, which would have no content.

We shouldn't treat that as a complete load.  We could also potentially use a URL white/blacklist, but URLs are changeable

File: lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutDialog.kt [hunk 2]
```diff
@@ -135,10 +133,7 @@ internal class CheckoutDialog(
 
     private fun hideProgressBar() {
         findViewById<FrameLayout>(R.id.checkoutSdkLoadingSpinner).visibility = GONE
-        findViewById<CheckoutWebViewContainer>(R.id.checkoutSdkContainer).apply {
-            children.firstOrNull()?.scrollY = 0
```
Comment (cianBuckley) [inline lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutDialog.kt:139]: Is this no longer needed? It doesn't seem related?

File: lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt (hunk 1 above)
Comment (markmur) [reply lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt:141]: Could you check the response status for a 302 instead?

Comment (kiftio) [reply lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt:141]: I found it strange yeah

Copying a URL into chrome and checking the network tab, I see these though, which match up:
```
Request URL: https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/cart/c/c1-e42cf7c4a06bb118769277249677d426?key=f00f2b671b20f8fa4c5af55b1430067f
Request Method: GET
Status Code: 302 Found
```

```
Request URL: https://shop1.shopify.checkout-web-fvy7.daniel-kift.eu.spin.dev/checkouts/cn/c1-e42cf7c4a06bb118769277249677d426
Request Method: GET
Status Code: 200 OK
```

Comment (markmur) [reply lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt:141]: Is it not possible to extract the status code from the webview somehow?

File: lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutDialog.kt (hunk 2 above)
Comment (kiftio) [reply lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutDialog.kt:139]: moving to a separate PR
